npm/webpack-dev-server/src/helpers/angularHandler.ts

```diff
@@ -0,0 +1,237 @@
+import * as fs from 'fs-extra'
+import { tmpdir } from 'os'
+import * as path from 'path'
+import { pathToFileURL } from 'url'
+import type { PresetHandlerResult, WebpackDevServerConfig } from '../devServer'
+import { sourceDefaultWebpackDependencies } from './sourceRelativeWebpackModules'
+
+export type AngularJsonProjectConfig = {
+  projectType: string
+  root: string
+  sourceRoot: string
+  architect: {
+    build: {
+      options: { [key: string]: any } & { polyfills?: string }
+      configurations?: {
+        [configuration: string]: {
+          [key: string]: any
+        }
+      }
+    }
+  }
+}
+
+type AngularJson = {
+  defaultProject?: string
+  projects: {
+    [project: string]: AngularJsonProjectConfig
+  }
+}
+
+const dynamicImport = new Function('specifier', 'return import(specifier)')
+
+export async function angularHandler (devServerConfig: WebpackDevServerConfig): Promise<PresetHandlerResult> {
+  const webpackConfig = await getAngularCliWebpackConfig(devServerConfig)
+
+  return { frameworkConfig: webpackConfig, sourceWebpackModulesResult: sourceDefaultWebpackDependencies(devServerConfig) }
+}
+
+async function getAngularCliWebpackConfig (devServerConfig: WebpackDevServerConfig) {
+  const { projectRoot } = devServerConfig.cypressConfig
+
+  const {
+    generateBrowserWebpackConfigFromContext,
+    getCommonConfig,
+    getStylesConfig,
+  } = await getAngularCliModules(projectRoot)
+
+  const angularJson = await getAngularJson(projectRoot)
+
+  let { defaultProject } = angularJson
+
+  if (!defaultProject) {
+    defaultProject = Object.keys(angularJson.projects).find((name) => angularJson.projects[name].projectType === 'application')
+
+    if (!defaultProject) {
+      throw new Error('Could not find a project with projectType "application" in "angular.json"')
+    }
+  }
+
+  const defaultProjectConfig = angularJson.projects[defaultProject]
+
+  const tsConfig = await generateTsConfig(devServerConfig, defaultProjectConfig)
+
+  const buildOptions = getAngularBuildOptions(defaultProjectConfig, tsConfig)
+
+  const context = createFakeContext(projectRoot, defaultProject, defaultProjectConfig)
+
+  const { config } = await generateBrowserWebpackConfigFromContext(
+    buildOptions,
+    context,
+    (wco: any) => [getCommonConfig(wco), getStylesConfig(wco)],
+  )
+
+  delete config.entry.main
+
+  return config
+}
+
+export function getAngularBuildOptions (projectConfig: AngularJsonProjectConfig, tsConfig: string) {
+  return {
```

Comment:
These options are derived from https://github.com/angular/angular-cli/blob/main/packages/angular_devkit/build_angular/src/builders/browser/schema.json

A comment would be useful here.